### PR TITLE
fix changelog validation

### PR DESCRIPTION
**Pull request by dzinad**

### CODE REVIEW

**Comment by kmadsen** [comment]
am attempting to run the tests but get a warning
```
➜  scripts git:(dd-fix-changelog-validation) ✗ python3 -m unittest validate-changelog-tests.py
........../Users/kylemadsen/Development/mapbox/mapbox-navigation-android/scripts/validate-changelog-tests.py:649: ResourceWarning: unclosed file <_io.TextIOWrapper name='/Users/kylemadsen/Development/mapbox/mapbox-navigation-android/scripts/test_resources/test_changelog.md' mode='r' encoding='UTF-8'>
  return open(abs_file_path, "r").read()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
..../Users/kylemadsen/Development/mapbox/mapbox-navigation-android/scripts/validate-changelog-tests.py:649: ResourceWarning: unclosed file <_io.TextIOWrapper name='/Users/kylemadsen/Development/mapbox/mapbox-navigation-android/scripts/test_resources/test_changelog_single_line_repeats_stable_version.md' mode='r' encoding='UTF-8'>
  return open(abs_file_path, "r").read()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
.../Users/kylemadsen/Development/mapbox/mapbox-navigation-android/scripts/validate-changelog-tests.py:649: ResourceWarning: unclosed file <_io.TextIOWrapper name='/Users/kylemadsen/Development/mapbox/mapbox-navigation-android/scripts/test_resources/test_changelog_single_line_repeats_unstable_version.md' mode='r' encoding='UTF-8'>
  return open(abs_file_path, "r").read()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
....................
----------------------------------------------------------------------
Ran 37 tests in 0.011s

OK
```

Here's a fix, it seems the `with` block will close the file after it is read
```
    def read_test_changelog(self, filename):
        script_dir = os.path.dirname(__file__)
        rel_path = "test_resources/" + filename
        abs_file_path = os.path.join(script_dir, rel_path)
        with open(abs_file_path, 'r') as f:
            data = f.read()
        return data
```

https://stackoverflow.com/questions/21057942/python-unclosed-resource-is-it-safe-to-delete-the-file

File: scripts/validate-changelog-tests.py [hunk 1]
```diff
@@ -121,6 +124,150 @@ def test_extract_added_lines_only_changelog_has_added(self):
         '''
         self.assertEqual(validate_changelog_utils.extract_added_lines(diff), expected)
 
+    def test_extract_added_lines_only_changelog_with_path_has_added(self):
+        expected = {
+            "libnavui-androidauto/CHANGELOG.md" : ['- Added third unreleased feature. [#6050](https://github.com/mapbox/mapbox-navigation-android/pull/6050)']
```
Comment (kmadsen) [inline scripts/validate-changelog-tests.py:129]: ❤️

File: scripts/validate-changelog-tests.py [hunk 2]
```diff
@@ -121,6 +124,150 @@ def test_extract_added_lines_only_changelog_has_added(self):
...
+        }
+        diff = '''diff --git a/libnavui-androidauto/CHANGELOG.md b/libnavui-androidauto/CHANGELOG.md
```
Comment (kmadsen) [inline scripts/validate-changelog-tests.py:131]: nitpick: should this be a file in the `test_resources`?

**Review by kmadsen — APPROVED** [review]

File: scripts/validate-changelog-tests.py (hunk 2 above)
Comment (dzinad) [reply scripts/validate-changelog-tests.py:131]: Well, these diffs are not reusable and IMO not too long to be placed on a separate file. The present files are used twice or more.

**Pull request merged**